src/Microsoft.AspNetCore.Mvc.Abstractions/ModelBinding/ModelStateDictionary.cs

```diff
@@ -37,8 +40,8 @@ public ModelStateDictionary()
         public ModelStateDictionary(int maxAllowedErrors)
         {
             MaxAllowedErrors = maxAllowedErrors;
-
-            _innerDictionary = new Dictionary<string, ModelStateEntry>(StringComparer.OrdinalIgnoreCase);
+            var emptySegment = new StringSegment(buffer: string.Empty);
+            _root = new ModelStateNode(parent: null, key: emptySegment, subKey: emptySegment);
```

Comment:
FYI stringsegment is bigger than string, and can "hide" a `ToString()` if you're not careful. I'd rather see `key` represented as a string. Subkey is fine as `StringSegment`.